core/state/statedb.go

```diff
@@ -834,25 +1193,53 @@ func (s *StateDB) StorageTrie(addr common.Address) Trie {
 }
 
 func (s *StateDB) HasSuicided(addr common.Address) bool {
+	if s.parallel.isSlotDB {
+		// 1.Try to get from dirty
+		if obj, ok := s.parallel.dirtiedStateObjectsInSlot[addr]; ok {
+			log.Info("HasSuicided in dirty", "addr", addr, "suicided", obj.suicided)
+			return obj.suicided
+		}
+		// 2.Try to get from uncomfirmed
+		if obj := s.getAddrStateFromUnconfirmedDB(addr); obj != nil {
+			log.Info("HasSuicided in unconfirmed", "addr", addr, "suicided", obj.suicided)
+			return obj.suicided
+		}
+	}
 	stateObject := s.getStateObject(addr)
 	if stateObject != nil {
+		log.Info("HasSuicided in main", "addr", addr, "suicided", stateObject.suicided)
 		return stateObject.suicided
 	}
+	log.Info("HasSuicided not exist", "addr", addr)
 	return false
 }
 
 /*
  * SETTERS
  */
 
+// the source obj is got from UnConfirmedDB in slot or the main StateDB
+// the obj should have reliable nonce, but un-reliable balance, code and state(deleted, suicided)
+func (s *StateDB) unconfirmedLightCopy(obj *StateObject) *StateObject {
+	newObj := obj.lightCopy(s)
+	// do balance fixup
+	if balance := s.getBalanceFromUnconfirmedDB(obj.address); balance != nil {
```

Comment:
```
		newStateObject := s.unconfirmedLightCopy(stateObject)
		newStateObject.AddBalance(amount)
		s.parallel.dirtiedStateObjectsInSlot[addr] = newStateObject
```

since we will set `dirtiedStateObjectsInSlot` after `unconfirmedLightCopy`, why `getBalanceFromUnconfirmedDB` since it will always return nil.

`if obj, exist := db.parallel.dirtiedStateObjectsInSlot[addr]; exist {`